pylint/checkers/variables.py

```diff
@@ -662,6 +661,48 @@ def get_next_to_consume(self, node):
             self.consumed_uncertain[node.name] += difference
             found_nodes = filtered_nodes
 
+        # If this node is in a Finally block of a Try/Finally,
+        # filter out assignments in the try portion, assuming they may fail
+        if (
+            found_nodes
+            and isinstance(node.statement(future=True).parent, nodes.TryFinally)
```

Comment:
Let's store `node.statement` in a local variable. We're calling it 4/5 times in this block.